docs/content/tutorial/step_06.ngdoc

```diff
@@ -1,108 +1,239 @@
+the model.
+
+
+# Testing
+
+The changes we made should be verified with both a unit test and an E2E test. Let's look at the unit
+test first.
+
+<br />
+**`app/phone-list/phone-list.component.spec.js`:**
+
+```js
+describe('phoneList', function() {
+
+  // Load the module that contains the `phoneList` component before each test
+  beforeEach(module('phoneList'));
+
+  // Test the controller
+  describe('PhoneListController', function() {
+    var ctrl;
+
+    beforeEach(inject(function($componentController) {
+      ctrl = $componentController('phoneList', {$scope: {}});
+    }));
+
+    it('should create a `phones` model with 3 phones', function() {
+      expect(ctrl.phones.length).toBe(3);
+    });
+
+    it('should set a default value for the `orderProp` model', function() {
+      expect(ctrl.orderProp).toBe('age');
     });
-...
+
+  });
+
+});
+```
+
+The unit test now verifies that the default ordering property is set.
+
+We used Jasmine's API to extract the controller construction into a `beforeEach` block, which is
+shared by all tests in the parent `describe` block.
+
+You should now see the following output in the Karma tab:
+
+```
+Chrome 49.0: Executed 2 of 2 SUCCESS (0.136 secs / 0.08 secs)
+```
+
+Let's turn our attention to the E2E tests.
+
+<br />
+**`e2e-tests/scenarios.js`:**
+
+```js
+  describe('PhoneCat Application', function() {
+
+    describe('phoneList', function() {
+
+      ...
+
+      it('should be possible to control phone order via the drop-down menu', function() {
+        var phoneNameColumn = element.all(by.repeater('phone in $ctrl.phones').column('phone.name'));
+        var query = element(by.model('$ctrl.query'));
+
+        function getNames() {
+          return phoneNameColumn.map(function(elem) {
+            return elem.getText();
+          });
+        }
+
+        query.sendKeys('tablet'); // Let's narrow the dataset to make the test assertions shorter
+
+        expect(getNames()).toEqual([
+          'Motorola XOOM\u2122 with Wi-Fi',
+          'MOTOROLA XOOM\u2122'
+        ]);
+
+        element(by.model('$ctrl.orderProp')).element(by.css('option[value="name"]')).click();
+
+        expect(getNames()).toEqual([
+          'MOTOROLA XOOM\u2122',
+          'Motorola XOOM\u2122 with Wi-Fi'
+        ]);
+      });
+
+      ...
 ```
 
-We added a new end-to-end test to verify that the app is generating correct links to the phone
-views that we will implement in the upcoming steps.
+The E2E test verifies that the ordering mechanism of the select box is working correctly.
 
 You can now rerun `npm run protractor` to see the tests run.
 
 
 # Experiments
 
-* Replace the `ng-src` directive with a plain old `src` attribute. Using tools such as Firebug,
-or Chrome's Web Inspector, or inspecting the webserver access logs, confirm that the app is indeed
-making an extraneous request to `/app/%7B%7Bphone.imageUrl%7D%7D` (or
-`/app/{{phone.imageUrl}}`).
+<div></div>
```

Comment:
`<ul>` right after a heading (`# Experiments`) have some different styling. Inserting a `<div>` to avoid `h1 + ul` selectors (iirc). Not ideal, I know :smiley: